RoadNamer/Utilities/ThreadingMonitor.cs

```diff
@@ -0,0 +1,33 @@
+﻿using ICities;
+using System.Collections.Generic;
+
+namespace RoadNamer.Managers
+{
+    public class ThreadingMonitor : ThreadingExtensionBase
+    {
+        public override void OnCreated(IThreading threading)
+        {
+            base.OnCreated(threading);
+
+        }
+
+        public override void OnAfterSimulationTick()
+        {
+            HashSet<ushort> segmentsToBeDeleted = new HashSet<ushort>();
```

Comment:
Should add a comment here ( i.e:After simulating, check segments that have a name, and remove the name if the underlying segment no longer exists )